src/transformers/pipelines/image_to_text.py

```diff
@@ -98,9 +105,34 @@ def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Imag
         """
         return super().__call__(images, **kwargs)
 
-    def preprocess(self, image):
+    def preprocess(self, image, prompt=None):
         image = load_image(image)
         model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+
+        if self.model.config.model_type == "git" and prompt is None:
+            model_inputs["input_ids"] = None
+
+        if prompt is not None and not isinstance(prompt, str):
+            raise ValueError(
+                f"Received an invalid text input, got - {type(prompt)} - but expected a single string. "
+                "Note also that one single text can be provided for conditional image to text generation."
+            )
+
+        elif prompt is not None and self.model.config.model_type == "git":
+            input_ids = self.tokenizer(text=prompt, add_special_tokens=False).input_ids
+            input_ids = [self.tokenizer.cls_token_id] + input_ids
```

Comment:
This is odd.

why `add_special_tokens=False` and then readd them right aftewards  ?
This could become an issue on finetuned model that might use different prompt tokens directly from the configuration.